File: Collection assignment/src/com/assignment/school/Student.java
package com.assignment.school;

public class Student extends Person{
    private int studentID;
    private int studentGrade;
    private static int tempID=1;

    public Student(String name, int age,int grade) {
        super(name, age);
        this.studentID = tempID;
        this.studentGrade =grade;
        tempID++;
    }

    public int getStudentGrade() {
        return studentGrade;
    }

    public void setStudentGrade(int studentGrade) {
        this.studentGrade = studentGrade;
    }

    public int getStudentID() {
        return studentID;
    }

    public void setStudentID(int studentID) {
        this.studentID = studentID;
    }

    public static int getTempID() {
        return tempID;
    }

    public static void setTempID(int tempID) {
        Student.tempID = tempID;
    }

    @Override
    public String toString() {
        return "Student{" +"studentName="+super.getName()+" studentAge"+super.getAge()+
                " studentID=" + studentID +" studentGrade"+ studentGrade +
                '}';
    }
}

File: Frontend/Typescript/Inheritance/singlelevel/Student.ts
import { Person } from "./Person";

class Student extends Person{
    id:number;
    grade:number;

    override talk(): void {
        console.log("Im a student");
    }

}

let student = new Student;
student.name="mithul";
student.age=14;
student.gender="male";
student.id = 12201;
student.grade = 8

console.log(student);
student.talk();